cassovary-core/src/test/scala/com/twitter/cassovary/util/SmallBoundedPriorityQueueSpec.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.util
+
+import org.scalatest.WordSpec
+import org.scalatest.matchers.ShouldMatchers
+
+class SmallBoundedPriorityQueueSpec extends WordSpec with ShouldMatchers {
+  "SmallBoundedPriorityQueue of 3 ints" when {
+    val q = new SmallBoundedPriorityQueue[Int](3)
```

Comment:
the problem with doing this is that these tests can not be done in parallel because of shared mutable state held by the instance val. see http://doc.scalatest.org/2.2.1/#org.scalatest.WordSpec and search for "Note that the only way before and after code can communicate with test code".

One option is to change this to a 'def fixture = new ...' and inserting 'val q = fixture' at the beginning of each "in" block